hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/ReadFilesPTest.java

```diff
@@ -121,10 +148,78 @@ private Pipeline buildDag(String glob) {
         return p;
     }
 
-    private void finishDirectory(File ... files) {
+    private Pipeline pipelineJson() {
+        Pipeline p = Pipeline.create();
+        p.readFrom(Sources.filesBuilder(directory.getPath())
+                          .buildJson(TestPerson.class))
+         .writeTo(Sinks.list(listJson));
+
+        return p;
+    }
+
+    private void finishDirectory(File... files) {
         for (File file : files) {
             assertTrue(file.delete());
         }
         assertTrue(directory.delete());
     }
+
+    public static class TestPerson implements Serializable {
```

Comment:
Maybe use Jackson jr annotation support to get rid of all those getters & setters?